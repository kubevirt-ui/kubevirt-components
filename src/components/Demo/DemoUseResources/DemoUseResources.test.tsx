import * as React from 'react';

import { cleanup, render } from '@testing-library/react';

import { DemoUseResources } from './DemoUseResources';

afterEach(cleanup);

const name = 'vm-example';
const namespace = 'default';
const errorText = 'Missing virtual machine data';
const dataTestID = 'DemoUseResourceTest';

test('Render DemoUseResources (valid vm)', () => {
  const { asFragment } = render(
    <DemoUseResources
      name={name}
      namespace={namespace}
      errorText={errorText}
      dataTestID={dataTestID}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Render DemoUseResources (none valid vm)', () => {
  const { asFragment } = render(
    <DemoUseResources
      name="none"
      namespace="does-not-exist"
      errorText={errorText}
      dataTestID={dataTestID}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
