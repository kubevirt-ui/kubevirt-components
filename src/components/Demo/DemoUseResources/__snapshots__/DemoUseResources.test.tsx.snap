// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DemoUseResources (none valid vm) 1`] = `
<DocumentFragment>
  <small
    class=""
    data-ouia-component-id="OUIA-Generated-Text-1"
    data-ouia-component-type="PF4/Text"
    data-ouia-safe="true"
    data-pf-content="true"
    data-test-id="DemoUseResourceTest"
  >
    Missing virtual machine data
  </small>
</DocumentFragment>
`;

exports[`Render DemoUseResources (valid vm) 1`] = `
<DocumentFragment>
  <div
    class="pf-c-label-group"
  >
    <div
      class="pf-c-label-group__main"
    >
      <ul
        aria-label="Label group category"
        class="pf-c-label-group__list"
        role="list"
      >
        <li
          class="pf-c-label-group__list-item"
        >
          <span
            class="pf-c-label"
          >
            <a
              class="pf-c-label__content"
              href="#"
            >
              <span
                class="pf-c-label__text"
              >
                VMINotExists=False
              </span>
            </a>
          </span>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
